wcslib doesn't compile

tried on Ubuntu 18.04 with gcc@7.3.0 and centos 7 with gcc@6.4.0

```
$ cd spack
$  git describe 
v0.8.17-8602-g51bd319fc
$ bin/spack install wcslib
==> Installing wcslib
==> Using cached archive: /home/gijs/Work/spack/var/spack/cache/wcslib/wcslib-5.18.tar.bz2
==> Staging archive: /home/gijs/Work/spack/var/spack/stage/wcslib-5.18-z7fhkbdz2cgbvkq62hjjtgoil67ltqsi/wcslib.tar.bz2
==> Created stage in /home/gijs/Work/spack/var/spack/stage/wcslib-5.18-z7fhkbdz2cgbvkq62hjjtgoil67ltqsi
==> No patches needed for wcslib
==> Building wcslib [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j16' 'install'

45 errors found in build log:
     505            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     506     ar r libwcs-PIC.a spc.o
     507     
     508     /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcserr.c
     509     ar r libwcs-PIC.a lin.o
     510     ar r libwcs-PIC.a spx.o
  >> 511     ar: lin.o: No such file or directory
     512     GNUmakefile:131: recipe for target 'libwcs-PIC.a(lin.o)' failed
  >> 513     make[1]: *** [libwcs-PIC.a(lin.o)] Error 1
     514     
     515     /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcsfix.c
     516     
     517     /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcshdr.c
     518     ar r libwcs-PIC.a wcserr.o
     519     

     ...

     732            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     733     ar r libwcs-5.18.a wcs.o
     734     make[1]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'wcs.o' will always be updated.
              Please consider passing the U flag to ar to avoid the problem.
     735     ar r libwcs-PIC.a wcsutrn.o
     736     make[1]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcsutrn.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     737     ar r libwcs-PIC.a dis.o
  >> 738     ar: dis.o: No such file or directory
     739     GNUmakefile:131: recipe for target 'libwcs-PIC.a(dis.o)' failed
  >> 740     make[1]: *** [libwcs-PIC.a(dis.o)] Error 1
     741     ar r libwcs-PIC.a wcsulex.o
     742     make[1]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcsulex.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     743     ar r libwcs-5.18.a fitshdr.o
     744     make[1]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'fitshdr.o' will always be upda
             ted. Please consider passing the U flag to ar to avoid the problem.
     745     ar r libwcs-5.18.a wcsbth.o
     746     ar r libwcs-PIC.a wcs.o
  >> 747     ar: wcs.o: No such file or directory
     748     GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcs.o)' failed
  >> 749     make[1]: *** [libwcs-PIC.a(wcs.o)] Error 1
     750     make[1]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'wcsbth.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     751     ar r libwcs-PIC.a wcshdr.o
  >> 752     ar: wcshdr.o: No such file or directory
     753     GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcshdr.o)' failed
  >> 754     make[1]: *** [libwcs-PIC.a(wcshdr.o)] Error 1
     755     ar r libwcs-PIC.a wcspih.o
  >> 756     ar: wcspih.o: No such file or directory
     757     GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcspih.o)' failed
  >> 758     make[1]: *** [libwcs-PIC.a(wcspih.o)] Error 1
     759     ar r libwcs-PIC.a wcsbth.o
  >> 760     ar: wcsbth.o: No such file or directory
     761     GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcsbth.o)' failed
  >> 762     make[1]: *** [libwcs-PIC.a(wcsbth.o)] Error 1
     763     ar r libwcs-5.18.a prj.o
     764     make[1]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'prj.o' will always be updated.
              Please consider passing the U flag to ar to avoid the problem.
     765     ar r libwcs-PIC.a prj.o
  >> 766     ar: prj.o: No such file or directory
     767     GNUmakefile:131: recipe for target 'libwcs-PIC.a(prj.o)' failed
  >> 768     make[1]: *** [libwcs-PIC.a(prj.o)] Error 1
     769     make[1]: Target 'build' not remade because of errors.
     770     make[1]: Leaving directory '/tmp/gijs/spack-stage/spack-stage-7G5fjm/wcslib-5.18/C'
     771     
     772     do 2018/06/28 11:34:51 +0200, executing on
     773     make[1]: Entering directory '/tmp/gijs/spack-stage/spack-stage-7G5fjm/wcslib-5.18/Fortran'
     774     

     ...

     1482              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     1483           __bos (__s), __fmt, __va_arg_pack ());
     1484           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     1485    ar r libwcs-5.18.a wcspih.o
     1486    make[2]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'wcspih.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     1487    ar r libwcs-PIC.a dis.o
  >> 1488    ar: dis.o: No such file or directory
     1489    GNUmakefile:131: recipe for target 'libwcs-PIC.a(dis.o)' failed
  >> 1490    make[2]: *** [libwcs-PIC.a(dis.o)] Error 1
     1491    ar r libwcs-PIC.a wcsutrn.o
     1492    make[2]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcsutrn.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     1493    ar r libwcs-5.18.a wcsbth.o
     1494    make[2]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'wcsbth.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     1495    ar r libwcs-PIC.a wcsulex.o
  >> 1496    ar: wcsulex.o: No such file or directory
     1497    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcsulex.o)' failed
  >> 1498    make[2]: *** [libwcs-PIC.a(wcsulex.o)] Error 1
     1499    make[2]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcsulex.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     1500    ar r libwcs-PIC.a wcs.o
  >> 1501    ar: wcs.o: No such file or directory
     1502    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcs.o)' failed
  >> 1503    make[2]: *** [libwcs-PIC.a(wcs.o)] Error 1
     1504    ar r libwcs-PIC.a wcshdr.o
  >> 1505    ar: wcshdr.o: No such file or directory
     1506    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcshdr.o)' failed
  >> 1507    make[2]: *** [libwcs-PIC.a(wcshdr.o)] Error 1
     1508    ar r libwcs-PIC.a wcspih.o
  >> 1509    ar: wcspih.o: No such file or directory
     1510    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcspih.o)' failed
  >> 1511    make[2]: *** [libwcs-PIC.a(wcspih.o)] Error 1
     1512    ar r libwcs-PIC.a wcsbth.o
  >> 1513    ar: wcsbth.o: No such file or directory
     1514    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcsbth.o)' failed
  >> 1515    make[2]: *** [libwcs-PIC.a(wcsbth.o)] Error 1
     1516    ar r libwcs-5.18.a prj.o
     1517    make[2]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'prj.o' will always be updated.
              Please consider passing the U flag to ar to avoid the problem.
     1518    ar r libwcs-PIC.a prj.o
  >> 1519    ar: prj.o: No such file or directory
     1520    GNUmakefile:131: recipe for target 'libwcs-PIC.a(prj.o)' failed
  >> 1521    make[2]: *** [libwcs-PIC.a(prj.o)] Error 1
     1522    make[2]: Target 'lib' not remade because of errors.
     1523    make[2]: Leaving directory '/tmp/gijs/spack-stage/spack-stage-7G5fjm/wcslib-5.18/C'
     1524    GNUmakefile:234: recipe for target '../C/libwcs-5.18.a' failed
  >> 1525    make[1]: *** [../C/libwcs-5.18.a] Error 2
     1526    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'tab_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.
     1527    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'wcsfix_f.o' will always b
             e updated. Please consider passing the U flag to ar to avoid the problem.
     1528    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'fitshdr_f.o' will always 
             be updated. Please consider passing the U flag to ar to avoid the problem.
     1529    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'log_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.
     1530    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'prj_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.
     1531    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'spx_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.

     ...

     2108    
     2109    /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcs.c
     2110    make[2]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcserr.o' will always be update
             d. Please consider passing the U flag to ar to avoid the problem.
     2111    
     2112    /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcsbth.c
     2113    ar r libwcs-PIC.a fitshdr.o
  >> 2114    ar: fitshdr.o: No such file or directory
     2115    GNUmakefile:131: recipe for target 'libwcs-PIC.a(fitshdr.o)' failed
  >> 2116    make[2]: *** [libwcs-PIC.a(fitshdr.o)] Error 1
     2117    make[2]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcsfix.o' will always be update
             d. Please consider passing the U flag to ar to avoid the problem.
     2118    
     2119    /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcserr.c
     2120    ar r libwcs-PIC.a lin.o
     2121    
     2122    /home/gijs/Work/spack/lib/spack/env/gcc/gcc -DHAVE_CONFIG_H -I. -I.. -march=native -O2 -pipe -fPIC -c wcsfix.c

     ...

     2319    /usr/include/x86_64-linux-gnu/bits/stdio2.h:33:10: note: ‘__builtin___sprintf_chk’ output between 3 and 74 bytes into a
              destination of size 72
     2320       return __builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,
     2321              ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     2322           __bos (__s), __fmt, __va_arg_pack ());
     2323           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     2324    ar r libwcs-PIC.a dis.o
  >> 2325    ar: dis.o: No such file or directory
     2326    GNUmakefile:131: recipe for target 'libwcs-PIC.a(dis.o)' failed
  >> 2327    make[2]: *** [libwcs-PIC.a(dis.o)] Error 1
     2328    wcspih.l: In function ‘wcspih_final’:
     2329    wcspih.l:2476:26: warning: ‘%d’ directive writing between 1 and 10 bytes into a region of size 9 [-Wformat-overflow=]
     2330               dpfill(disp->dp+(idp++), "DQ", field, i+1, 1, 0, wval);
     2331                              ^~~~~~~~~~~~~~
     2332    wcspih.l:2476:26: note: directive argument in the range [0, 2147483647]
     2333    In file included from /usr/include/stdio.h:862:0,

     ...

     2343    make[2]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'wcspih.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     2344    ar r libwcs-5.18.a wcsbth.o
     2345    ar r libwcs-PIC.a wcsulex.o
     2346    make[2]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'wcsbth.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     2347    make[2]: Warning: Archive 'libwcs-PIC.a' seems to have been created in deterministic mode. 'wcsulex.o' will always be updat
             ed. Please consider passing the U flag to ar to avoid the problem.
     2348    ar r libwcs-PIC.a wcshdr.o
  >> 2349    ar: wcshdr.o: No such file or directory
     2350    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcshdr.o)' failed
  >> 2351    make[2]: *** [libwcs-PIC.a(wcshdr.o)] Error 1
     2352    ar r libwcs-5.18.a prj.o
     2353    make[2]: Warning: Archive 'libwcs-5.18.a' seems to have been created in deterministic mode. 'prj.o' will always be updated.
              Please consider passing the U flag to ar to avoid the problem.
     2354    ar r libwcs-PIC.a wcs.o
  >> 2355    ar: wcs.o: No such file or directory
     2356    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcs.o)' failed
  >> 2357    make[2]: *** [libwcs-PIC.a(wcs.o)] Error 1
     2358    ar r libwcs-PIC.a wcspih.o
  >> 2359    ar: wcspih.o: No such file or directory
     2360    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcspih.o)' failed
  >> 2361    make[2]: *** [libwcs-PIC.a(wcspih.o)] Error 1
     2362    ar r libwcs-PIC.a wcsbth.o
  >> 2363    ar: wcsbth.o: No such file or directory
     2364    GNUmakefile:131: recipe for target 'libwcs-PIC.a(wcsbth.o)' failed
  >> 2365    make[2]: *** [libwcs-PIC.a(wcsbth.o)] Error 1
     2366    ar r libwcs-PIC.a prj.o
  >> 2367    ar: prj.o: No such file or directory
     2368    GNUmakefile:131: recipe for target 'libwcs-PIC.a(prj.o)' failed
  >> 2369    make[2]: *** [libwcs-PIC.a(prj.o)] Error 1
     2370    make[2]: Target 'lib' not remade because of errors.
     2371    make[2]: Leaving directory '/tmp/gijs/spack-stage/spack-stage-7G5fjm/wcslib-5.18/C'
     2372    GNUmakefile:234: recipe for target '../C/libwcs-5.18.a' failed
  >> 2373    make[1]: *** [../C/libwcs-5.18.a] Error 2
     2374    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'tab_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.
     2375    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'wcsfix_f.o' will always b
             e updated. Please consider passing the U flag to ar to avoid the problem.
     2376    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'fitshdr_f.o' will always 
             be updated. Please consider passing the U flag to ar to avoid the problem.
     2377    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'sph_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.
     2378    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'log_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.
     2379    make[1]: Warning: Archive '../C/libwcs-5.18.a' seems to have been created in deterministic mode. 'prj_f.o' will always be u
             pdated. Please consider passing the U flag to ar to avoid the problem.

     ...

     2425       fi
     2426    /usr/bin/install -c -m 644 fitshdr.1 HPXcvt.1 wcsware.1 /home/gijs/Work/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/
             wcslib-5.18-z7fhkbdz2cgbvkq62hjjtgoil67ltqsi/share/man/man1
     2427    make[1]: Leaving directory '/tmp/gijs/spack-stage/spack-stage-7G5fjm/wcslib-5.18/utils'
     2428    /usr/bin/install -c -m 444 wcsconfig.h wcsconfig_f77.h /home/gijs/Work/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/w
             cslib-5.18-z7fhkbdz2cgbvkq62hjjtgoil67ltqsi/include/wcslib-5.18
     2429    /usr/bin/install: target '/home/gijs/Work/spack/opt/spack/linux-ubuntu18.04-x86_64/gcc-7.3.0/wcslib-5.18-z7fhkbdz2cgbvkq62h
             jjtgoil67ltqsi/include/wcslib-5.18' is not a directory
     2430    GNUmakefile:80: recipe for target 'install' failed
  >> 2431    make: *** [install] Error 1

See build log for details:
  /home/gijs/Work/spack/var/spack/stage/wcslib-5.18-z7fhkbdz2cgbvkq62hjjtgoil67ltqsi/wcslib-5.18/spack-build.out
```

that seems to pass the build step, but then it fails on `install`:

```
==> Installing wcslib
==> Using cached archive: /home/molenaar/spack/var/spack/cache/wcslib/wcslib-5.18.tar.bz2
==> Staging archive: /home/molenaar/spack/var/spack/stage/wcslib-5.18-f7lxplaqilkpxixqfsfskesrpvggqp2v/wcslib_5.18.orig.tar.bz2
==> Created stage in /home/molenaar/spack/var/spack/stage/wcslib-5.18-f7lxplaqilkpxixqfsfskesrpvggqp2v
==> No patches needed for wcslib
==> Building wcslib [AutotoolsPackage]
==> Executing phase: 'autoreconf'
==> Executing phase: 'configure'
==> Executing phase: 'build'
==> Executing phase: 'install'
==> Error: ProcessError: Command exited with status 2:
    'make' 'install'

2 errors found in build log:
     663       fi
     664    /usr/bin/install -c -m 444 wcslib.pdf /home/molenaar/spack/opt/spack/linux-centos7-x86_64/gcc-6.4.0/wcslib-5.18-f7lxplaqilkp
            xixqfsfskesrpvggqp2v/share/doc/wcslib-5.18
     665    if [ ! -d "/home/molenaar/spack/opt/spack/linux-centos7-x86_64/gcc-6.4.0/wcslib-5.18-f7lxplaqilkpxixqfsfskesrpvggqp2v/share/
            doc/wcslib-5.18/html" ] ; then \
     666         /usr/bin/install -c -d -m 775 /home/molenaar/spack/opt/spack/linux-centos7-x86_64/gcc-6.4.0/wcslib-5.18-f7lxplaqilkpxix
            qfsfskesrpvggqp2v/share/doc/wcslib-5.18/html ; \
     667       fi
     668    /usr/bin/install -c -m 444 html/* /home/molenaar/spack/opt/spack/linux-centos7-x86_64/gcc-6.4.0/wcslib-5.18-f7lxplaqilkpxixq
            fsfskesrpvggqp2v/share/doc/wcslib-5.18/html
  >> 669    /usr/bin/install: cannot stat ‘html/*’: No such file or directory
  >> 670    make: *** [install] Error 1

```

--- a/var/spack/repos/builtin/packages/wcslib/package.py
+++ b/var/spack/repos/builtin/packages/wcslib/package.py
@@ -38,9 +38,9 @@ class Wcslib(AutotoolsPackage):
     variant('x',       default=False, description='Use the X Window System')
 
     depends_on('gmake', type='build')
-    depends_on('flex',  type='build')
+    depends_on('flex@2.5.9:', type='build')
     depends_on('cfitsio', when='+cfitsio')
-    depends_on('libx11',  when='+x')
+    depends_on('libx11', when='+x')
 
     def configure_args(self):
         spec = self.spec

--- a/var/spack/repos/builtin/packages/wcslib/package.py
+++ b/var/spack/repos/builtin/packages/wcslib/package.py
@@ -29,7 +29,39 @@ class Wcslib(AutotoolsPackage):
     """WCSLIB a C implementation of the coordinate transformations
     defined in the FITS WCS papers."""
 
-    homepage = "http://www.atnf.csiro.au/people/mcalabre/WCS/"
-    url      = "ftp://ftp.atnf.csiro.au/pub/software/wcslib/wcslib.tar.bz2"
+    homepage = "http://www.atnf.csiro.au/people/mcalabre/WCS/wcslib/"
+    url      = "ftp://ftp.atnf.csiro.au/pub/software/wcslib/wcslib-5.18.tar.bz2"
 
     version('5.18', '67a78354be74eca4f17d3e0853d5685f')
+
+    variant('cfitsio', default=False, description='Include CFITSIO support')
+    variant('x',       default=False, description='Use the X Window System')
+
+    depends_on('gmake', type='build')
+    depends_on('flex',  type='build')
+    depends_on('cfitsio', when='+cfitsio')
+    depends_on('libx11',  when='+x')
+
+    def configure_args(self):
+        spec = self.spec
+
+        # TODO: Add PGPLOT package
+        args = ['--without-pgplot']
+
+        if '+cfitsio' in spec:
+            args.extend([
+                '--with-cfitsio',
+                '--with-cfitsiolib={0}'.format(
+                    spec['cfitsio'].libs.directories[0]),
+                '--with-cfitsioinc={0}'.format(
+                    spec['cfitsio'].headers.directories[0]),
+            ])
+        else:
+            args.append('--without-cfitsio')
+
+        if '+x' in spec:
+            args.append('--with-x')
+        else:
+            args.append('--without-x')
+
+        return args

